### PR TITLE
PR3 Delivery - GRP 2

### PR DESCRIPTION
**Pull request by ElectroNugget**
This is the delivery for PR3. The app works in its current state with some significant styling and backend work still to be done. 
All values for emissions are placeholder at the moment, but illustrate that the flow works. View the README for more info.

@mircealungu @madsnedergaard @mortenbjoern @Akons97

### CODE REVIEW

File: README.md
Comment (madsnedergaard) [inline README.md:12]: You should not keep this list in a README - when you run `npm still PACKAGE_NAME` it will automatically be added to the list of dependencies in `package.json`.

File: src/Main/App.js
Comment (madsnedergaard) [inline src/Main/App.js]: I guess this setup kinda works, but using a router would have made things easier - and made it possible to refresh the page without being sent back to the start (and logged out again?) :)

File: src/Pages/Dashboard/Dashboard.js
Comment (madsnedergaard) [inline src/Pages/Dashboard/Dashboard.js:10]: What's the reason for having firebase in the UserContext?

File: src/Pages/Dashboard/Depricated/Pledges/Pledge.js
Comment (madsnedergaard) [inline src/Pages/Dashboard/Depricated/Pledges/Pledge.js]: If this code is deprecated it should just be removed.

File: src/Pages/Dashboard/FootprintSlideshow/PieChart.js
Comment (madsnedergaard) [inline src/Pages/Dashboard/FootprintSlideshow/PieChart.js]: It would indeed 😉 
This should do the trick, you can repeat the same thing from transport on the other items:

```js
  const { transport: { sumTotal: transportTotal }, food, household, goods } = footprint;
```

File: src/Pages/Dashboard/PledgesPage/PledgesSection.js
Comment (madsnedergaard) [inline src/Pages/Dashboard/PledgesPage/PledgesSection.js]: You should only use `let` if the variable if you intend the variable to change later - in this case you don't :)

Screenshot 2020-11-13 at 21 25 22

File: src/Pages/Dashboard/Pledges/Pledges.js
Comment (madsnedergaard) [inline src/Pages/Dashboard/Pledges/Pledges.js]: This pattern of listing multiple variables on the same "line" (with just one var/let/const) is generally considered bad practice these days :)

Here's a few articles that describes why:
- https://eslint.org/docs/rules/one-var
- https://www.breck-mckye.com/blog/2017/01/the-javascript-single-var-style-is-an-antipattern/

File: src/Pages/Dashboard/Stats/Stats.js
Comment (madsnedergaard) [inline src/Pages/Dashboard/Stats/Stats.js]: Two things here:
- Try to use `classes` over `id`s - there can only be one element with the same id, so if you want to suddenly reuse some styling you will have to change this again
- Naming of things - I have no idea what these abbreviations mean, my guess would by "average monthly output", but the text says yearly. You could instead just call it `className="wrapper"` or something like that?

**Review by madsnedergaard — CHANGES_REQUESTED** [review]
Nice work here! 💪 

I have some comments here that I'd like you to address before this is merged in :)

File: src/Main/App.js
Comment (bwesth) [reply src/Main/App.js]: Yeah. We are sorta working on that. Debating whether we have the time atm. It will probably be added closer to the final handin deadline.

File: src/Pages/Dashboard/Dashboard.js
Comment (Rasmus-adn) [reply src/Pages/Dashboard/Dashboard.js:10]: so we can use the Firebase API where we need it without importing the firebase.js. Since the UserProvider UserContext wraps the whole App in index.js. Is this a b ad approach, should we rather just import it where needed?

Comment (bwesth) [reply src/Pages/Dashboard/Dashboard.js:10]: How would you do it?

File: src/Main/App.js
Comment (Rasmus-adn) [reply src/Main/App.js]: Mads do you think its worth the effort refactoring to use router at this point?? We talked about it and at this point it seems like it would require a mayor refactoring for little improvement?

File: README.md
Comment (bwesth) [reply README.md:12]: Fair.

File: src/Pages/Dashboard/Depricated/Pledges/Pledge.js
Comment (bwesth) [reply src/Pages/Dashboard/Depricated/Pledges/Pledge.js]: Didn't want to delete stuff until it was completely replaced. We are ready to delete at this point.

File: src/Pages/Dashboard/FootprintSlideshow/PieChart.js
Comment (bwesth) [reply src/Pages/Dashboard/FootprintSlideshow/PieChart.js]: Yeah, we were almost there, but it was late Wedensday. Thanks for the example :)

File: src/Pages/Dashboard/PledgesPage/PledgesSection.js
Comment (bwesth) [reply src/Pages/Dashboard/PledgesPage/PledgesSection.js]: Typo. Cheers :)

File: src/Pages/Dashboard/Pledges/Pledges.js
Comment (bwesth) [reply src/Pages/Dashboard/Pledges/Pledges.js]: Meh! Will read articles. I like doing it, but I will not do it when working with others from now on.

File: src/Pages/Dashboard/Stats/Stats.js
Comment (bwesth) [reply src/Pages/Dashboard/Stats/Stats.js]: Fair.

**Comment by bwesth** [comment]
All comments resolved. What now @madsnedergaard?

File: src/Main/App.js
Comment (mircealungu) [inline src/Main/App.js]: loggedIn[0] is obscure.
what else is in that loggedIn array? 
if it were a dictionary, you could at least give a name to the variable.

File: src/Main/UserContext.js
Comment (mircealungu) [inline src/Main/UserContext.js]: can be rewritten with a dictionary, no? {"transport":0, "food":3, "good":6, etc...} . or?

File: src/Main/UserContext.js
Comment (mircealungu) [inline src/Main/UserContext.js]: Array(10).fill(4)

File: src/Pages/Dashboard/FootprintSlideshow/BarChart.js
Comment (mircealungu) [inline src/Pages/Dashboard/FootprintSlideshow/BarChart.js]: do you use prettier? this is strange formatting.

File: src/Pages/Dashboard/FootprintSlideshow/LineGraph.js
Comment (mircealungu) [inline src/Pages/Dashboard/FootprintSlideshow/LineGraph.js]: a bit too repetitive? 
remember, you're in the automation business ;)

File: src/Pages/Dashboard/_Dashboard.scss
Comment (mircealungu) [inline src/Pages/Dashboard/_Dashboard.scss:12]: cool that you're using scss and variables.

File: src/Pages/Frontpage/Survey/Depricated/Option.js
Comment (mircealungu) [inline src/Pages/Frontpage/Survey/Depricated/Option.js]: you can use destructuring to make your code more readable, e.g. export default ({text, themeIndex, optionIndex})

File: src/Pages/Frontpage/Survey/Depricated/Option.js
Comment (mircealungu) [inline src/Pages/Frontpage/Survey/Depricated/Option.js]: can you do that? :)

File: src/Main/App.js
Comment (madsnedergaard) [reply src/Main/App.js]: Guess it depends on what else you need to work on - you might be able to make it without, but could also end up spending a lot of time trying to make "work around" this problem. 

But yeah, it could make sense to postpone this since it's kinda working now, and then focus on your requirements first - then later if you have time/need for it you can do this :)

File: src/Pages/Dashboard/Dashboard.js
Comment (madsnedergaard) [reply src/Pages/Dashboard/Dashboard.js:10]: Generally it's a bit weird to put everything in context (and in the same one), especially a library - if it works for now you can keep it though, but otherwise consider moving it out :)

File: src/Data/survey.js
Comment (madsnedergaard) [inline src/Data/survey.js]: :cry:

File: src/Data/survey.js
Comment (Akons97) [inline src/Data/survey.js:4]: You have to update picture names so they start with capital letters. I had to do this manually in order to start the app

File: src/Pages/Frontpage/Frontpage.js
Comment (Akons97) [inline src/Pages/Frontpage/Frontpage.js:10]: Transition is not user friendly. When I click 'Test your CO2 footprint' I'm not aware that anything happens, just that I lose footer, so intuitively I scroll down and see the questions.

File: src/Pages/Frontpage/SignUpForm.js
Comment (Akons97) [inline src/Pages/Frontpage/SignUpForm.js]: type should be password so that text is marked with dots, not visible to others as all password fields are implemented that way.

**Review by Akons97 — CHANGES_REQUESTED** [review]
The problems I addressed are mainly quick fixes but will significantly improve your usability.

File: src/Pages/Frontpage/Survey/Depricated/Option.js
Comment (bwesth) [reply src/Pages/Frontpage/Survey/Depricated/Option.js]: Called a ghost div :)

File: src/Main/App.js
Comment (bwesth) [reply src/Main/App.js]: The array is a React State. useState() always returns an array of [state value, state changefunction]. loggedIn[0] returns the state value. Would it be alright simply to comment that loggedIn is a useState?

File: src/Main/UserContext.js
Comment (bwesth) [reply src/Main/UserContext.js:54]: By dictionary you mean object?

File: src/Pages/Dashboard/FootprintSlideshow/BarChart.js
Comment (bwesth) [reply src/Pages/Dashboard/FootprintSlideshow/BarChart.js]: Must have missed one.

File: src/Main/UserContext.js
Comment (bwesth) [reply src/Main/UserContext.js]: new Array(12).fill(4)
Sure. All those ES6 niceties, nom nom.

File: src/Main/UserContext.js
Comment (bwesth) [reply src/Main/UserContext.js]: Sure.

    const indexReference = {
      "transport": 0,
      "food": 3,
      "good": 6,
      "household": 9
    }
    let surveyResult = 0;
    let index = indexReference[theme];
    
    I personally don't find it more readable than a switch, but as you wish.

File: src/Pages/Dashboard/FootprintSlideshow/LineGraph.js
Comment (bwesth) [reply src/Pages/Dashboard/FootprintSlideshow/LineGraph.js]: We are. It is.
This was copied from victory examples site and not used, as we could not in due time make a scatter graph using trees instead of circles. The component is currently unused and will be deleted unless an epiphany or surprise wormhole a couple of weeks back in time occurs.

File: src/Pages/Dashboard/_Dashboard.scss
Comment (bwesth) [reply src/Pages/Dashboard/_Dashboard.scss:12]: Nicolai's handiwork :)
We found out later that we really shouldn't have bothered, and just used component styling >.<

File: src/Pages/Frontpage/Survey/Depricated/Option.js
Comment (bwesth) [reply src/Pages/Frontpage/Survey/Depricated/Option.js]: We are actually doing this in some places. I guess we should be consistent in our format.

File: src/Data/survey.js
Comment (bwesth) [reply src/Data/survey.js]: Missed one. We are using prettier.

File: src/Pages/Frontpage/Survey/Depricated/Option.js
Comment (bwesth) [reply src/Pages/Frontpage/Survey/Depricated/Option.js]: Ghost div. copies elements into the parent without creating a new div. Lets you cheat the rule of returning only 1 element. Not a good thing to do as a rule. Should be done only in select instances. It does help the designer as there are less layers of divs to worry about.

File: src/Pages/Frontpage/Frontpage.js
Comment (bwesth) [reply src/Pages/Frontpage/Frontpage.js:10]: We are aware. Have a resource to do this without manipulating the dom, just haven't gotten around to it.

File: src/Data/survey.js
Comment (bwesth) [reply src/Data/survey.js:4]: I suspect this is a bug in Git.
We had a different convention in the team as to file naming. Settled on all lower case and correctly edited both file names and imports. Some of us had to manually correct the files on our local version, even if the correct code was pushed and pulled. I think Git may not check capital letters of .png files when looking to see what to update on local during a pull.
Only conclusion I can think of given what we experienced while sitting next to each other physically.

Not sure what we can do about it.

File: src/Pages/Frontpage/SignUpForm.js
Comment (bwesth) [reply src/Pages/Frontpage/SignUpForm.js]: Should probably have commented on this in the code.
This was intentionally type text so that we could see what we wrote during testing.
Will be changed to password for final version.

**Review by madsnedergaard — APPROVED** [review]
Approving so you can proceed :)

**Pull request merged**
